kufar/books/service.go

```diff
@@ -49,6 +52,32 @@ func ReadAll(d Database) func(w http.ResponseWriter, r *http.Request) {
 //Update ...
 func Update(d Database) func(w http.ResponseWriter, r *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
+		//r.Header.Set("WWW-Authenticate", `Basic realm="Restricted"`)
+
+		//username and password not gotten :(
+		uname, pword, ok := r.BasicAuth()
+
+		//cant get the gredentials :(
+		b64 := r.Header.Get("Authorization")
+
+		s := strings.Split(b64, " ")
+		fmt.Println(s)
+		sDec, err := base64.StdEncoding.DecodeString(s[1])
+		if err != nil {
+			http.Error(w, http.StatusText(http.StatusInternalServerError)+": could not decode base64 string", http.StatusInternalServerError)
+			return
+		}
+
+		cred := strings.Split(string(sDec), ":")
+		if string(cred[0]) != uname && string(cred[1]) != pword {
```

Comment:
You are doing it right way, but you are comparing data that always be equal. Because one came from `BasicAuth` and second from header

Reply:
I thought we'll have to compare the data from the `BasicAuth` and  from the `Authorization` header
I can compare the data from the database. but that'll be in the `SignInUser` function.

Reply:
true...now I see it